### PR TITLE
Re-org export code

### PR DESCRIPTION
**Pull request by oyilmaz-nvidia**
# What does this PR do ?

Re-org the export code to make it more step by step. In the new code, there will be 3 main parts including the reading the nemo ckpt, converting nemo ckpt to trt-llm ckpt, and building the trt-llm engine.

Once this PR is completed, then the converting nemo ckpt to trt-llm ckpt, and building the trt-llm engine parts will be moved to the mcore.

### CODE REVIEW

**Review by jiemingz — APPROVED** [review]
lgtm!

**Review by JimmyZhang12 — DISMISSED** [review]

**Review by meatybobby — APPROVED** [review]
LGTM

File: nemo/export/trt_llm/nemo_ckpt_loader/nemo_file.py [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+        nemo_dir = Path(nemo_ckpt)
+    else:
+        nemo_dir = TarPath(nemo_ckpt)
+
+    try:
+        unpacked_checkpoint_dir = UnpackedNemoCheckpointDir(nemo_dir, load_checkpoints_to_cpu=True)
+
+        dist_ckpt_folder = nemo_dir / "model_weights"
+        if dist_ckpt_folder.exists():
+            model = load_sharded_metadata(dist_ckpt_folder)
```
[the remark this answers is not in the document]
Comment (oyilmaz-nvidia) [reply nemo/export/trt_llm/nemo_ckpt_loader/nemo_file.py:232]: Yeah we still get the state_dict and this also works model_state_dict = model.get("state_dict", model). I'll update this when we pass the nemo model directly.

File: nemo/export/trt_llm/nemo/nemo_ckpt_convert.py [hunk 2]
```diff
@@ -1,647 +0,0 @@
...
-from pathlib import Path
-from typing import Union
-
-import numpy as np
-import tensorstore  # This is important even though not used. Otherwise zarr raises error.
-import torch
-import zarr
-from tensorrt_llm._utils import np_bfloat16, pad_vocab_size, str_dtype_to_torch, torch_to_numpy
-from torch.distributed.checkpoint import FileSystemReader, TensorStorageMetadata
-from torch.distributed.checkpoint.state_dict_loader import load_state_dict
```
Comment (mikolajblaz) [inline nemo/export/trt_llm/nemo/nemo_ckpt_convert.py:33]: @oyilmaz-nvidia I can see a few imports are gone in the new `nemo_file`, leading to NameErrors: `TensorStorageMetadata`, `load_state_dict`, `np_bfloat16`. Can you bring them back please?

(Also `GPT2Tokenizer` is missing but I don't know if this one is intended)

**Pull request merged**